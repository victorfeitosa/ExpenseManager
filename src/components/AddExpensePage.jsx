import React from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const AddExpensePage = (props) => (
  <div>
    <h1>Add expenses</h1>
    <ExpenseForm
      buttonText="Add expense"
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense));
        props.history.push('/');
      }} />
  </div>
);

export default connect()(AddExpensePage);