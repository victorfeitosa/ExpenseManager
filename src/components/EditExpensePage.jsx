import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.startRemoveExpense({id: this.props.expense.id});
    this.props.history.push('/');
  };

  render () {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          buttonText="Edit expense"
          onSubmit={this.onSubmit}>
        </ExpenseForm>
        <button onClick={this.onRemove}>Remove Expense &times;</button>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, updates) => dispatch(startEditExpense(id, updates)),
  startRemoveExpense: ({id}) => dispatch(startRemoveExpense({id}))
});

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((e) => e.id === props.match.params.id)
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
