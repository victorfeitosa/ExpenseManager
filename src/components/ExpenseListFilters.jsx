import React from 'react';
import { DateRangePicker } from 'react-dates';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  render () {
    return (
      <div>
        <input type="text" value={this.props.filters.text} onChange={(e) => {
          this.props.dispatch(setTextFilter(e.target.value));
        }}/>
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') this.props.dispatch(sortByDate());
            else this.props.dispatch(sortByAmount());
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="filter_start_date"
          endDate={this.props.filters.endDate}
          endDateId="filter_end_date"
          focusedInput={this.state.calendarFocused}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          onDatesChange={this.onDatesChange}
          onFocusChange={this.onFocusChange}
        >
        </DateRangePicker>
      </div>);
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
});

export default connect(mapStateToProps)(ExpenseListFilters);
