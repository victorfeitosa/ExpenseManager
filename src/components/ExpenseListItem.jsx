import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <p>{description}</p>
    <b>{amount}</b>
    <p>{createdAt}</p>

    <Link to={'/edit/' + id.toString()}>Edit</Link>
  </div>
);

export default ExpenseListItem;
