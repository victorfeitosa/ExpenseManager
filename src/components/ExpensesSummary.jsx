import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expensesTotal';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
   const expensesWord = expenseCount === 1 ? 'expense' : 'expenses';
   return (
      <div className='page-header'>
         <div className='content-container'>
            <h1 className='page-header__title'>Viewing <span>{expenseCount}</span> {expensesWord} totalling <span>{numeral(expensesTotal / 100).format('$0,0.00')}</span></h1>
            <div className='page-header__actions'>
               <Link className='button' to='/create'>Add expense</Link>
            </div>
         </div>
      </div>
   )
}

const mapStateToProps = (state) => {
   const visibleExpenses = selectExpenses(state.expenses, state.filters);
   return {
      expenseCount: visibleExpenses.length,
      expensesTotal: expensesTotal(visibleExpenses)
   }
};

export default connect(mapStateToProps)(ExpensesSummary);
