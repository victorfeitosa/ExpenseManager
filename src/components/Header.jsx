import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

export const Header = (props) => (
   <header>
      {
         true &&
         <div>
            <h1>Expense Manager</h1>
            <nav>
               <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
               <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
               <NavLink to="/help" activeClassName="is-active">Help</NavLink>
            </nav>
         </div>
      }
   </header>
);

// const mapStateToProps = (state) => ({
//    isLoggedIn: state.app.loggedIn
// });

export default connect()(Header);
