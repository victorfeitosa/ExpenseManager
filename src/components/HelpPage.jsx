import React from 'react';

const HelpPage = () => (
   <p>HELP</p>
);
export default HelpPage;
