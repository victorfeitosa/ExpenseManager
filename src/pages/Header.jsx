import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
   <header>
      <h1>Expense Manager</h1>
      <nav>
         <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
         <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
         <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </nav>
   </header>
);
export default Header;
