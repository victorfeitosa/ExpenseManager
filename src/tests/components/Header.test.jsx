import React from 'react';
import { Header } from '../../components/Header';

import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header startLogout={() => {}}/>);
  expect(toJson(wrapper)).toMatchSnapshot();
});
