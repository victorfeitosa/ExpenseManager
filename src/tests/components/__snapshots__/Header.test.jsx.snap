// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <div>
    <h1>
      Expense Manager
    </h1>
    <nav>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Home
      </NavLink>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/create"
      >
        Create expense
      </NavLink>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/help"
      >
        Help
      </NavLink>
    </nav>
  </div>
</header>
`;
